stdlib/public/core/String.swift

```diff
@@ -205,20 +205,20 @@ extension String {
   /// - Parameter encoding: describes the encoding in which the code units
   ///   should be interpreted.
   public init<Encoding: UnicodeEncoding>(
-    cString nulTerminatedCodeUnits: UnsafePointer<Encoding.CodeUnit>,
-    encoding: Encoding.Type) {
+    decodingCString nulTerminatedCodeUnits: UnsafePointer<Encoding.CodeUnit>,
```

Comment:
Is `decodingCString` significantly better than just `cString`? Seems like it would be nice for it to match the one that doesn't take an encoding, similar to if the encoding were a default argument.